File: src/main/java/com/example/storage/controller/dto/EquipmentDto.java
package com.example.storage.controller.dto;

import com.example.storage.entity.Equipment;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class EquipmentDto {
    private long id;
    private String name;
    private String inventoryCode;
    private String category;

    public static EquipmentDto toDto(Equipment equipment) {
        return new EquipmentDto(
                equipment.getId(),
                equipment.getName(),
                equipment.getInventoryCode(),
                equipment.getCategory().getName()
        );
    }
}

File: src/main/kotlin/ru/nto/storage/controller/dto/EquipmentDto.kt
package ru.nto.storage.controller.dto

import ru.nto.storage.entity.Equipment

data class EquipmentDto(
    val id: Long,
    val name: String,
    val inventoryCode: String,
    val category: String
) {
    companion object {
        fun toDto(equipment: Equipment) = EquipmentDto(
            id = equipment.id,
            name = equipment.name,
            inventoryCode = equipment.inventoryCode,
            category = equipment.category.name
        )
    }
}
